rstudio/rstudio
# Posit cloud option flag

### Intent

Addresses [rstudio-pro#3994](https://github.com/rstudio/rstudio-pro/issues/3994)
Also addresses [#12241](https://github.com/rstudio/rstudio/issues/12241)

### Approach

Untangle logic between a new Enable Cloud publishing UI from the pre-existing Enable RStudio (Posit) Connect publishing UI (renamed the option while I'm at it)

Check behavior of that option in the new account wizards as well as the Rmd publishing wizard 

### Automated Tests

None

### QA Notes

For effective testing, we should test 
1) When publishing to Connect is enabled and publishing to Cloud is disabled (the default)
2) When publishing to Connect is disabled and publishing to Cloud is enabled
3) When publishing to both Connect and Cloud are disabled

Option 3 should test with a shiny app and an rmarkdown doc
Options 1 and 2 should test with a shiny app, a plumber api, and an rmarkdown doc, ideally

Option 4 with both publishing options enabled can be tested, but I think will be redundant
